src/vehicle.cpp

```diff
@@ -5393,7 +5393,7 @@ void vehicle::open_or_close(int part_index, bool opening)
       int ydiff = parts[next_index].mount_dy - parts[part_index].mount_dy;
       if((xdiff * xdiff + ydiff * ydiff == 1) && // (x^2 + y^2) == 1
               (part_info(next_index).id == part_info(part_index).id) &&
-              (parts[next_index].open == opening ? 0 : 1)) {
+              (parts[next_index].open == 1 && opening)) {
```

Comment:
If `opening` is false, the condition will not be true ever. This is not the same behavior as the current code. The current code checks a *relation* between `.open` and `opening`. The changed code does not check that relation, it check their state completely separated.